keras_cv/layers/patch_merging.py

```diff
@@ -0,0 +1,66 @@
+"""
+Code copied and modified from
+https://github.com/rwightman/pytorch-image-models/blob/master/timm/models/swin_transformer.py
+https://github.com/sayakpaul/swin-transformers-tf/blob/main/swins/layers/patch_merging.py
+"""
+
+from functools import partial
+
+import tensorflow as tf
+from tensorflow.keras import layers
+
+class PatchMerging(layers.Layer):
+    """Patch Merging Layer.
+    Args:
+        input_resolution (tuple[int]): Resolution of input feature.
+        dim (int): Number of input channels.
+    """
+
+    def __init__(
+        self,
+        input_resolution,
+        dim,
+        out_dim=None,
+        norm_layer=partial(layers.LayerNormalization, epsilon=1e-5),
+        **kwargs
+    ):
+        super().__init__(**kwargs)
+
+        self.input_resolution = input_resolution
+        self.dim = dim
+        self.out_dim = out_dim or 2 * dim
+        self.norm = norm_layer()
+        self.reduction = layers.Dense(self.out_dim, use_bias=False)
+
+    def call(self, x):
+        """
+        x: B, H*W, C
+        """
+        H, W = self.input_resolution
+        B, L, C = tf.shape(x)[0], tf.shape(x)[1], tf.shape(x)[2]
```

Comment:
refactor this to use a single `tf.shape(x)` call.  `tf.shape()` gets encoded in the graph as an op, so its more performant to assign to an intermediate `input_shape` variable.

Reply:
Great call!